api/src/opentrons/protocol_api/module_contexts.py

```diff
@@ -425,70 +415,33 @@ def engage(
         """
         if height is not None:
-            dist = height
+            self._core.engage(height_from_home=height)
 
         # This version check has a bug:
         # if the caller sets height_from_base in an API version that's too low,
         # we will silently ignore it instead of raising APIVersionError.
         # Leaving this unfixed because we haven't thought through
         # how to do backwards-compatible fixes to our version checking itself.
         elif height_from_base is not None and self._api_version >= APIVersion(2, 2):
-            dist = (
-                height_from_base
-                + modules.magdeck.OFFSET_TO_LABWARE_BOTTOM[self._module.model()]
-            )
-
-        elif self.labware and self.labware.magdeck_engage_height is not None:
-            dist = self._determine_lw_engage_height()
-            if offset:
-                dist += offset
+            self._core.engage(height_from_base=height_from_base)
 
         else:
-            raise ValueError(
-                "Currently loaded labware {} does not have a known engage "
-                "height; please specify explicitly with the height param".format(
-                    self.labware
-                )
+            self._core.engage_to_labware(
+                offset=offset or 0,
+                preserve_half_mm=self._api_version < APIVersion(2, 3),
             )
 
-        self._module.engage(dist)
-
-    def _determine_lw_engage_height(self) -> float:
-        """Return engage height based on Protocol API and module versions
-
-        For API Version 2.3 or later:
-           - Multiply non-standard labware engage heights by 2 for gen1 modules
-           - Divide standard labware engage heights by 2 for gen2 modules
-        If none of the above, return the labware engage heights as defined in
-        the labware definitions
-        """
-        assert self.labware
-        assert self.labware.magdeck_engage_height
-
-        engage_height = self.labware.magdeck_engage_height
-
-        is_api_breakpoint = self._api_version >= APIVersion(2, 3)
-        is_v1_module = self._module.model() == "magneticModuleV1"
-        engage_height_is_in_half_mm = self.labware.load_name in MAGDECK_HALF_MM_LABWARE
-
-        if is_api_breakpoint and is_v1_module and not engage_height_is_in_half_mm:
-            return engage_height * ENGAGE_HEIGHT_UNIT_CNV
-        elif is_api_breakpoint and not is_v1_module and engage_height_is_in_half_mm:
-            return engage_height / ENGAGE_HEIGHT_UNIT_CNV
-        else:
-            return engage_height
-
     @publish(command=cmds.magdeck_disengage)
     @requires_version(2, 0)
     def disengage(self) -> None:
         """Lower the magnets back into the Magnetic Module."""
-        self._module.deactivate()
+        self._core.disengage()
 
     @property  # type: ignore
     @requires_version(2, 0)
     def status(self) -> str:
         """The status of the module; either 'engaged' or 'disengaged'"""
```

Comment:
```suggestion
        """The status of the module: either ``engaged`` or ``disengaged``"""
```
Double backticks for sphinx monospace formatting.